src/data/users.js

```diff
@@ -78,6 +78,44 @@ const addRole = async (id, role) => {
     { returnDocument: "after" }
   );
 };
+
 const removeRole = async (id, role) => {};
 
-export { createUser, updateUser, addRole, findUser, findUserByEmailOrUserName };
+const getAllExperts = async () => {
+  try {
+    const db = await users();
+    const experts = await db.find({ role: 'expert' }).toArray();
+    return experts;
+  } catch (e) {
+    throw new DataBaseException(e);
+  }
+};
+
+const getExpertById = async (userId) => {
+  try {
+    const db = await users();
+    const expert = await db.findOne({ _id: new ObjectId(userId), role: 'expert' });
+    if (!expert) throw new DataBaseException(`Expert with ID ${userId} not found`);
+    return expert;
+  } catch (e) {
+    throw new DataBaseException(e);
+  }
+};
+
+const searchExpertsByName = async (name) => {
+  try {
```

Comment:
same, add validation function, validate name.
you can just validate if its string by first null check, then name.checkString()